Remove name column from default ransackable attributes

mdesantis opened the pull request:
Fixes #3179. `name` column included by default within default ransackable attributes causes ransack searches with search matchers based on `name` column to crash when appied on models which don’t have a `name` column. This PR removes the `name` column from default ransackable attributes and adds it explicitly on models which support search by name.

**Checklist:**
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- [ ] I have updated Guides and README accordingly to this change (if needed)
- [ ] I have added tests to cover this change (if needed)

mdesantis commented:
@kennyadsl do you think this is an important change? because previous implementation could expect `default_ransack_attributes` to include `name`? I was thinking about that

kennyadsl commented:
`Important` label is needed to mark PRs that need special attention in the Changelog. I think this could potentially break user application if they were relying on having that `name` attribute in the ranskack attributes, so they have to be aware that, in case they really need it, they have to re-add it manually, right?

mdesantis commented:
@kennyadsl , in this PR I fixed the tested models, but if we want to be :100:% sure that the functionality remains the same for the remaining models we could add `name` to their `whitelisted_ransackable_attributes` too. Here is the models list which have a `name` column but doesn't have `name` within `whitelisted_ransackable_attributes`:

```ruby
Spree::AdjustmentReason
Spree::BillingIntegration
Spree::CreditCard
Spree::Gateway
Spree::Gateway::Bogus
Spree::Gateway::BogusSimple
Spree::PaymentMethod
Spree::PaymentMethod::BogusCreditCard
Spree::PaymentMethod::Check
Spree::PaymentMethod::CreditCard
Spree::PaymentMethod::SimpleBogusCreditCard
Spree::PaymentMethod::StoreCredit
Spree::PromotionCategory
Spree::RefundReason
Spree::ReimbursementType
Spree::ReimbursementType::Credit
Spree::ReimbursementType::Exchange
Spree::ReimbursementType::OriginalPayment
Spree::ReimbursementType::StoreCredit
Spree::ReturnReason
Spree::Role
Spree::ShippingCategory
Spree::ShippingMethod
Spree::StateChange
Spree::Store
Spree::StoreCreditCategory
Spree::StoreCreditReason
Spree::StoreCreditType
Spree::TaxCategory
Spree::TaxRate
```

I extracted this list with the following script:

```ruby
ActiveRecord::Base.descendants.select do |model|
  begin
    model.respond_to?(:whitelisted_ransackable_attributes) && 
    model.columns.any? { |column| column.name == 'name' } && (
      !model.whitelisted_ransackable_attributes ||
      !model.whitelisted_ransackable_attributes.include?('name')
    )
  rescue ActiveRecord::StatementInvalid; false
  end
end.map(&:to_s).sort
```
